yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtTree.kt

```diff
@@ -597,22 +592,39 @@ internal class CrdtTreeNode(
     /**
      * Copies itself deeply.
      */
-    fun deepcopy(): CrdtTreeNode {
-        val clone = CrdtTreeNode(pos, type, _attributes = _attributes.deepCopy())
-        val node = this
-        return clone.apply clone@{
-            removedAt = node.removedAt
-            if (node.isText) {
-                value = node.value
+    fun deepCopy(): CrdtTreeNode {
+        return CrdtTreeNode(
+            pos,
+            type,
+            value.takeIf { isText },
+            _children.map { child ->
+                child.deepCopy()
+            }.toMutableList(),
+            _attributes.deepCopy(),
+        ).also {
+            it.size = size
+            it.removedAt = removedAt
+            it._children.forEach { child ->
+                child.parent = it
             }
-            size = node.size
-            node._children.map { child ->
-                child.deepcopy().apply {
-                    parent = this@clone
-                }
-            }.let(_children::addAll)
         }
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (other !is CrdtTreeNode) {
+            return false
+        }
+        if (this === other) {
+            return true
+        }
+        return pos == other.pos && type == other.type && value == other.value &&
+            attributes == other.attributes && createdAt == other.createdAt &&
```

Comment:
I think `createdAt == other.createdAt` is redundant because `CrdtTreeNode.createdAt` is equal to `CrdtTreeNode.pos.createdAt` and it's already checking `pos == other.pos`.